tests/test_sourmash.py

```diff
@@ -72,6 +72,77 @@ def test_sourmash_info_verbose():
     assert "loaded from path" in err
 
 
+def test_load_pathlist_from_file_does_not_exist():
+    from sourmash.sourmash_args import load_pathlist_from_file
+    with pytest.raises(ValueError) as e:
+        load_pathlist_from_file("")
+    assert "file '' does not exist" in str(e.value)
+
+
+@utils.in_tempdir
+def test_load_pathlist_from_file_empty(c):
+    from sourmash.sourmash_args import load_pathlist_from_file
+    file_list = c.output("file_list")
+    with open(file_list, "w") as fp:
+        fp.write("")
+    with pytest.raises(ValueError) as e:
+        load_pathlist_from_file(file_list)
+    assert "pathlist is empty" in str(e.value)
+
+
+@utils.in_tempdir
+def test_load_pathlist_from_file_badly_formatted(c):
+    from sourmash.sourmash_args import load_pathlist_from_file
+    file_list = c.output("file_list")
+
+    with open(file_list, "w") as fp:
+        fp.write("{'a':1}")
+
+    with pytest.raises(ValueError) as e:
+        load_pathlist_from_file(file_list)
+    assert "pathlist contains a badly formatted file" in str(e.value)
+    
+
+@utils.in_tempdir
+def test_load_pathlist_from_file_badly_formatted_2(c):
+    from sourmash.sourmash_args import load_pathlist_from_file
+    file_list = c.output("file_list")
+    
+    sig1 = utils.get_test_data('compare/genome-s10.fa.gz.sig')
+    newdir = c.output('newdir')
+    os.mkdir(newdir)
+    new_file = os.path.join(newdir, 'sig1')
+    shutil.copyfile(sig1, new_file)
+
+    with open(file_list, "w") as fp:
+        fp.write(new_file + "\n")
+        fp.write("{'a':1}")
+
+    with pytest.raises(ValueError) as e:
+        load_pathlist_from_file(file_list)
+    assert "pathlist contains a badly formatted file" in str(e.value)
+
+
+@utils.in_tempdir
+def test_load_pathlist_from_file_duplicate(c):
+    from sourmash.sourmash_args import load_pathlist_from_file
+    file_list = c.output("file_list")
+
+    sig1 = utils.get_test_data('compare/genome-s10.fa.gz.sig')
+    newdir = c.output('newdir')
+    os.mkdir(newdir)
+    new_file = os.path.join(newdir, 'sig1')
+    shutil.copyfile(sig1, new_file)
+
+    with open(file_list, "w") as fp:
+        fp.write(new_file + "\n")
+        fp.write(new_file + "\n")
+
+    with pytest.raises(ValueError) as e:
+        load_pathlist_from_file(file_list)
+    assert "pathlist contains a duplicate file" in str(e.value)
```

Comment:
so, I still think we shouldn't care about duplicate files, and we should just remove them silently. See comment [here](https://github.com/dib-lab/sourmash/pull/1469#issuecomment-827224466) suggesting that we do that silently and return a set rather than a list from this function.

Reply:
to implement this, you could make `file_list` into a set immediately the list comprehension, and remove all mention of `duplicate`.

Reply:
So if I convert `file_list` into a set, will it take care of duplicates on its own? Should I do this conversion in `tests/test_sourmash.py` or should I do it inside the function itself in `src/sourmash/sourmash_args.py`?

Reply:
Should we return a set from `src/sourmash/sourmash_args.py` instead of a list? In this case, I would have to write the code as something like this:
`with open(filename, 'rt') as fp:`
      `temp_file_list = [ x.rstrip('\r\n') for x in fp ]`
`file_list = set(temp_file_list)`

I'm still not sure how to handle duplicate files. If I modify the code this way, should I not care about them at all? Should I remove all the code relating to the variable `duplicate` in `src/sourmash/sourmash_args.py`? And should I even remove the test case for the duplicate files in `tests/test_sourmash.py`?

Reply:
That should all work, yes. I think it's ok to silently ignore duplicate files.

I think having the test case for duplicate files is good, but it should verify that they are ignored instead of checking for the exception.

Reply:
regarding where to do the conversion -

the code in src is the application that is distributed, and that should have the behavior of silently collapsing and ignoring duplicates.

the tests check the behavior. they are not distributed with the application.

I suggest giving it a try and then asking for a review once it works ;).

Reply:
How can I test if the duplicates have been ignored?

Reply:
if you know how many distinct entries there are in the pathlist file, you can look at the return value of the `load_pathlist_from_file` function, no?